When describing classes in a Pipeline Shared Library that have "data" they must be serializable

rtyler opened the pull request:
Fixes [JENKINS-38868](https://issues.jenkins-ci.org/browse/JENKINS-38868)

bitwiseman commented:
🐝

bitwiseman commented on content/doc/book/pipeline/shared-libraries.adoc:
```diff
@@ -229,7 +229,7 @@ constructor, or just one method:
 ----
 package org.foo
-class Utilities {
+class Utilities implements Serializable {
```
Does this need updating as well? https://github.com/jenkins-infra/jenkins.io/pull/495/files#diff-d0909a306d46ea261898f36947fb9d9bL260

bitwiseman replied:
Hm, maybe not.  The other is static so it doesn't need to be serializable. 
On the other hand, the distinction seems like something that would be easy to mess up or forget if one goes from only having a static method to having some data.  It is worth highlighting with a warning.

rtyler replied:
I will try to add some verbiage around it, but I'm loathe to litter these documents with so many admonitions for every possible way to fuck up Groovy :slightly_frowning_face: :

bitwiseman commented:
https://github.com/jenkinsci/workflow-cps-global-lib-plugin/pull/23

bitwiseman commented:
🐝 Looks good to me.

rtyler commented:
Build failing right now because of

```
Could not resolve all dependencies for configuration ':nodeDist'.
110 	> Could not download node-linux-x64.tar.gz (org.nodejs:node:4.2.3)
111 	   > Could not get resource 'https://nodejs.org/dist/v4.2.3/node-v4.2.3-linux-x64.tar.gz'.
112 	      > Premature end of Content-Length delimited message body (expected: 11683079; received: 5078794
```

Cute

jglick commented:
Argh, we are writing *more* content that duplicates content defined somewhere else without deleting the original? Overlaps https://github.com/jenkinsci/workflow-cps-global-lib-plugin/pull/23. This is unmanageable. We need to have a policy for keeping a single version of the truth.

rtyler commented:
@jglick If you would like me to submit pull requests to various repositories when content is reworked for jenkins.io, I can do that. The documents aren't identical however, I removed and rewrote some content to try to make it more clear to readers as they progress through the chapter.
